components/supervisor/pkg/serverApi/publicapi.go

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package serverApi
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	backoff "github.com/cenkalti/backoff/v4"
+	configCat "github.com/gitpod-io/gitpod/common-go/experiments"
+	"github.com/gitpod-io/gitpod/common-go/log"
+	v1 "github.com/gitpod-io/gitpod/components/public-api/go/experimental/v1"
+	gitpod "github.com/gitpod-io/gitpod/gitpod-protocol"
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+)
+
+type ServerAPIInterface interface {
+	GetWorkspace(ctx context.Context, id string) (res *gitpod.WorkspaceInfo, err error)
+	GetToken(ctx context.Context, query *gitpod.GetTokenSearchOptions) (res *gitpod.Token, err error)
+	OpenPort(ctx context.Context, workspaceID string, port *gitpod.WorkspaceInstancePort) (res *gitpod.WorkspaceInstancePort, err error)
+	InstanceUpdates(ctx context.Context, instanceID string, workspaceID string) (<-chan *gitpod.WorkspaceInstance, error)
+
+	// Remove this and use segment client directly
+	TrackEvent(ctx context.Context, event *gitpod.RemoteTrackMessage) (err error)
+}
+
+const (
+	// KindGitpod marks tokens that provide access to the Gitpod server API.
+	KindGitpod = "gitpod"
+)
+
+type ServerApiService struct {
+	// cfg *Config
+	instanceID  string
+	workspaceID string
+
+	// gitpodService server API
+	gitpodService gitpod.APIInterface
+	// publicAPIConn public API publicAPIConn
+	publicAPIConn *grpc.ClientConn
+}
+
+var _ ServerAPIInterface = (*ServerApiService)(nil)
+
+func usePublicAPI(ctx context.Context, ownerID string, experiments configCat.Client) bool {
+	return ownerID != "" && experiments.GetBoolValue(ctx, "supervisor_experimental_publicapi", false, configCat.Attributes{
+		UserID: ownerID,
```

Comment:
maybe it should be instance id instead?

Reply:
we can use `ownerID` as user id for configcat, and add `teamID`

Reply:
WDYT? @akosyakov

Reply:
ok, let's go with `ownerID`